src/libraries/System.Private.CoreLib/src/System/IO/Strategies/Net5CompatFileStreamStrategy.Unix.cs

```diff
@@ -103,6 +104,28 @@ private void Init(FileMode mode, FileShare share, string originalPath, FileOptio
                     }
                 }
             }
+
+            // If allocationSize has been provided for a creatable and writeable file
+            if (allocationSize > 0 && (_access & FileAccess.Write) != 0 && mode != FileMode.Open && mode != FileMode.Append)
+            {
+                int fallocateResult = Interop.Sys.PosixFAllocate(_fileHandle, 0, allocationSize);
+                if (fallocateResult != 0)
+                {
+                    _fileHandle.Dispose();
```

Comment:
Seems to me that the catch in Net5Compat ctor already disposes the _fileHandle.

Reply:
Here I need to Dispose it before passing the path to `Unlink`, otherwise unlink is not going to remove the file immediately

Reply:
What could happen if it is not removed immediately?

Reply:
> What could happen if it is not removed immediately?

`Unlink` does not remove the file and on Linux, we end up having a full disk.